_posts/en/newsletters/2019-04-16-newsletter.md

```diff
@@ -0,0 +1,152 @@
+week.  You've probably seen us publish an occasional very long
+newsletter; this week, you see the opposite.)*
+
+## Bech32 sending support
+
+*Week 5 of 24.  Until the second anniversary of the segwit soft
+fork lock-in on 24 August 2019, the Optech Newsletter will contain this
+weekly section that provides information to help developers and
+organizations implement bech32 sending support---the ability to pay
+native segwit addresses.  This [doesn't require implementing
+segwit][bech32 easy] yourself, but it does allow the people you pay to
+access all of segwit's multiple benefits.*
+
+One reason your users and customers may want you to implement bech32
+sending support is because it'll allow the receivers of those payments
+to save on fees when they re-spend that money.  This week, we'll look at
+how much money they'll save and discuss how their savings could also
+help you save money.
+
+For the legacy P2PKH address format implemented in the first version of
+Bitcoin, the scriptSig that authorizes a spend is typically 107 vbytes.
+For P2SH-wrapped segwit P2WPKH, this same information is moved to a
+witness data field that only consumes 1/4 as many vbytes (27 vbytes) but
+whose P2SH overhead adds 23 vbytes for a total of 50 vbytes.  For native
+segwit P2WPKH, there's no P2SH overhead, so 27 vbytes is all that's used.
+
+This means you could argue that P2SH-P2WPKH saves over 50% compared to
+P2PKH, and that P2WPKH saves another almost 50% compared to P2SH-P2WPKH
+or 75% compared to P2PKH alone.  However, spending transactions contain
+more than just scriptSigs and witness data, so the way we usually
+compare savings is by looking at prototype transactions.  For example,
+we imagine a typical transaction containing a single input and two
+outputs (one to the receiver; one as change back to the spender).  In
+that case:
+
+- Spending P2PKH has a total transaction size of
+  220 vbytes
+- Spending P2SH-P2WPKH has a size of 167 vbytes (24% savings)
+- Spending P2WPKH output has a size of 141 vbytes (16% savings vs
+  P2SH-P2WPKH or 35% vs P2PKH)
+
+To compare simple multisig transactions (those that just use a single
+`OP_CHECKMULTSIG` opcode), things get more complex because k-of-n
+multisig inputs vary in size depending on the number of signatures (k)
+and the number of public keys (n).  So, for simplicity's sake, we'll
+just plot the sizes of legacy P2SH-multisig, wrapped P2SH-P2WSH
+multisig, and native P2WSH multisig for all the possible k-of-n multisig
+combinations (up to the maximum 15-of-15 supported by legacy P2SH).  We
+can see that switching to P2SH-P2WSH brings large savings.
+Further switching to native P2WSH brings an additional constant-sized
+savings (about 35 bytes) that can increase savings by up to 15% for
+small multisigs (but which is marginal for large multisigs).
+
+![Plot of multisig transaction sizes with P2SH, P2SH-P2WSH, and P2WSH](/img/posts/2019-04-segwit-multisig-size.png)
```

Comment:
I like this graph, but the P2SH-P2WSH plots are almost impossible to see. We could experiment with different layouts, or perhaps have two graphs. One for 1-of-n multisigs (for 1 <= n <= 15) and one for a k-of-15 graph (for 1 <= k <= 15).